### PR TITLE
efi: Improve EFI Compatibility

### PR DESCRIPTION
**Pull request by retrage**
This PR improves EFI compatibility to support Linux boot in upcoming aarch64 support tracked in issue #198.

### CODE REVIEW

File: src/efi/block.rs [hunk 1]
```diff
@@ -106,11 +106,14 @@ pub struct BlockWrappers<'a> {
     pub count: usize,
 }
 
-pub extern "win64" fn reset(_: *mut BlockIoProtocol, _: bool) -> Status {
+eficall! {
```
Comment (rbradford) [inline src/efi/block.rs:109]: This looks a bit weird to me but I guess rustfmt is happy with this!

**Review by rbradford — APPROVED** [review]

**Comment by retrage** [comment]
@rbradford I noticed that the use of `eficall!()` macro can be replaced with "efiapi" ABI. I pushed this change in [a75b253](https://github.com/cloud-hypervisor/rust-hypervisor-firmware/pull/204/commits/a75b2538a67c21ed265cb73e9154f5cbb710f9d0).

File: src/efi/block.rs (hunk 1 above)
Comment (retrage) [reply src/efi/block.rs]: Unfortunately, rustfmt is not happy with functions wrapped around macro because it cannot apply code format in the macro.

**Comment by rbradford** [comment]
> @rbradford I noticed that the use of `eficall!()` macro can be replaced with "efiapi" ABI. I pushed this change in [a75b253](https://github.com/cloud-hypervisor/rust-hypervisor-firmware/pull/204/commits/a75b2538a67c21ed265cb73e9154f5cbb710f9d0).

I think that looks a lot nicer! :rocket:

**Review by rbradford — APPROVED** [review]

**Pull request merged**